### PR TITLE
generate perm groups from clifford generators

### PR DESCRIPTION
**Pull request by stylewarning**

### CODE REVIEW

File: src/clifford/perm.lisp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+;;; This files deals with the permutation group representation of the
+;;; Clifford group.
+;;;
+;;; The permutation group representation is a direct consequence of
+;;; the definition of the Clifford group. Namely, the Clifford group
+;;; is the set of unitaries which stabilize the Pauli group under
+;;; conjugation. As such, we can just look at how a Clifford group
+;;; element permutes the Pauli group, and construct a homomorphism to
+;;; Z_n, where n ~ number of Paulis.
```
Comment (kilimanjaro) [inline src/clifford/perm.lisp]: Do you mean S_n?

File: src/clifford/perm.lisp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  ;; -i*P, +I, -I!
+  (- (/ (expt 4 (1+ n)) 2) 2))
+
+(defun clifford-to-perm (c)
+  "Convert a Clifford group element C into a permutation representation."
+  (let* ((num-qubits (num-qubits c))
+         (num-points (clifford-perm-degree num-qubits))
+         (rep (perm::iota-vector (1+ num-points))))
+    (dotimes (i num-points (perm::%make-perm :rep rep))
+      ;; The 2*(2 + i) and x/2 - 2 are to convert in/out from integers
```
Comment (kilimanjaro) [inline src/clifford/perm.lisp]: Make a function for this plz

File: src/clifford/pauli.lisp [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ;;;; src/clifford/pauli.lisp
 ;;;;
-;;;; Author: Nik Tezak
+;;;; Authors: Nik Tezak
+;;;;          Robert Smith
```
Comment (ecpeterson) [inline src/clifford/pauli.lisp]: stake ur claim

File: src/clifford/pauli.lisp [hunk 4]
```diff
@@ -80,6 +81,21 @@
         :do (setf idx (logior c (ash idx 2)))
         :finally (return idx)))
 
+(defun pauli-integer (p)
+  "Give a non-negative integer representation of the Pauli P, which is unique among only the Pauli operators of the same number of qubits."
```
Comment (ecpeterson) [inline src/clifford/pauli.lisp]: I'd like this docstring to say something about what happens to this indexing when you take the underlying collection of qubits and tensor by the identity on one qubit on the left & on the right.

File: src/clifford/perm.lisp (hunk 1 above)
Comment (stylewarning) [reply src/clifford/perm.lisp]: No, a Pauli is mapped to Z_n.

Comment (stylewarning) [reply src/clifford/perm.lisp]: yes

**Pull request merged**